modules/tinymce/src/core/main/ts/keyboard/Nbsps.ts

```diff
@@ -126,13 +136,16 @@ const normalizeNbspMiddle = (text: string): string => {
   }).join('');
 };
 
-const normalizeNbspAtStart = (root: SugarElement<Node>, node: Text): boolean => {
+const normalizeNbspAtStart = (root: SugarElement<Node>, node: Text, makeNbsb: boolean): boolean => {
   const text = node.data;
   const firstPos = CaretPosition(node, 0);
 
-  if (isNbspAt(text, 0) && !needsToBeNbsp(root, firstPos)) {
+  if (isNbspAt(text, 0) && !needsToBeNbsp(root, firstPos) && !makeNbsb) {
```

Comment:
It's probably worth putting the `makeNbsp` checks at the start of these if statements in this function and in `normalizeNbspAtEnd` below. As remember JavaScript will evaluate from left to right, so in this case it'll evaluate the more costly `needsToBeNbsp` when it's never going to be useful if `makeNbsp` is true anyways.